subproject/simple/simple_dragonbox.h

```diff
@@ -0,0 +1,1996 @@
+            static_assert(max_k - min_k <= 619 && compression_ratio == 27, "");
+            static_assert(max_k - min_k <= INT_MAX, "");
+            auto const cache_index = int(uint32_t(int(k - min_k) * int32_t(607)) >> 14);
+            auto const kb = int(cache_index * compression_ratio + min_k);
+            auto const offset = int(k - kb);
+
+            // Get the base cache.
+            auto const base_cache = cache[cache_index];
+
+            if (offset == 0) {
+                return base_cache;
+            }
+            else {
+                // Compute the required amount of bit-shift.
+                auto const alpha = int(floor_log2_pow10(k) - floor_log2_pow10(kb) - offset);
+                assert(alpha > 0 && alpha < 64);
+
+                // Try to recover the real cache.
+                auto const pow5 = pow5_table[offset];
+                auto recovered_cache = umul128(base_cache.high, pow5);
+                auto const middle_low = umul128(base_cache.low, pow5);
+
+                recovered_cache += middle_low.high;
+
+                auto const high_to_middle =
+                    uint64_t((recovered_cache.high << (64 - alpha)) & UINT64_C(0xffffffffffffffff));
+                auto const middle_to_low =
+                    uint64_t((recovered_cache.low << (64 - alpha)) & UINT64_C(0xffffffffffffffff));
+
+                recovered_cache = {(recovered_cache.low >> alpha) | high_to_middle,
+                                   ((middle_low.low >> alpha) | middle_to_low)};
+
+                assert(recovered_cache.low != UINT64_C(0xffffffffffffffff));
+                recovered_cache = {recovered_cache.high, uint64_t(recovered_cache.low + 1)};
+
+                return recovered_cache;
+            }
+        }
+    };
+
+    template <int a, class UInt>
+    constexpr int count_factors(UInt n) {
+        static_assert(a > 1);
+        int c = 0;
+        while (n % a == 0) {
+            n /= a;
+            ++c;
+        }
+        return c;
+    }
+
+    static constexpr uint32_t divide_magic_number[2]{6554, 656};
+
+    struct interval {
+        bool include_left_endpoint;
+        bool include_right_endpoint;
+    };
+
+    enum class binary_round_mode {
+        nearest_to_even,
+        nearest_to_odd,
+        nearest_toward_plus_infinity,
+        nearest_toward_minus_infinity,
+        nearest_toward_zero,
+        nearest_away_from_zero,
+        nearest_to_even_static_boundary,
+        nearest_to_odd_static_boundary,
+        nearest_toward_plus_infinity_static_boundary,
+        nearest_toward_minus_infinity_static_boundary,
+        toward_plus_infinity,
+        toward_minus_infinity,
+        toward_zero,
+        away_from_zero,
+    };
+
+    enum class decimal_round_mode {
+        to_even,
+        to_odd,
+        away_from_zero,
+        toward_zero,
+        dont_care,
+    };
+
+    static_assert(sizeof(float) == 4);
+    static_assert(sizeof(double) == 8);
```

Comment:
I think you can preserve more complex `static_assert` I did in the original implementation for these, with the error message. Ideally it would be great to have comments clarifying the restriction, but I can add that later.